ghcide/src/Development/IDE/Core/Shake.hs

```diff
@@ -842,18 +814,21 @@ defineEarlyCutoff op = addBuiltinRule noLint noIdentity $ \(Q (key, file)) (old
                 (bs, (diags, res)) <- actionCatch
                     (do v <- op key file; liftIO $ evaluate $ force v) $
                     \(e :: SomeException) -> pure (Nothing, ([ideErrorText file $ T.pack $ show e | not $ isBadDependency e],Nothing))
-                modTime <- liftIO $ (currentValue =<<) <$> getValues state GetModificationTime file
-                (bs, res) <- case res of
+                modTime <- liftIO $ (currentValue . fst =<<) <$> getValues state GetModificationTime file
+                (bs, diags, diagsV, res) <- case res of
```

Comment:
`diags` and `diagsV` are the same, and you just return both to avoid doing `Vector.fromList (Vector.toList v)`?

Reply:
Yes. Depending on the branch we start with either a list or a vector, and later we need both

Reply:
Instead can we make `updateFileDiagnostics` take a Vector?

Reply:
`updateFileDiagnostics` wants a list though, since it partitions diagnostics in shown and hidden

Reply:
Vector has partition operations too?

Reply:
Why would you want to allocate an intermediate vector though?